qfieldsync/core/cloud_api.py

```diff
@@ -182,19 +264,33 @@ def server_urls() -> List[str]:
             "https://localhost:8002/",
         ]
 
+    def get_username(self) -> Optional[str]:
+        if self.auth_method == CloudAuthMethod.CREDENTIALS:
+            return self.auth().config("username")
+        elif self.auth_method == CloudAuthMethod.SSO:
+            return self.current_username
+        return None
+
     def auth(self) -> QgsAuthMethodConfig:
         auth_manager = QgsApplication.authManager()
 
         if not auth_manager.masterPasswordHashInDatabase():
             return QgsAuthMethodConfig()
 
         authcfg = self.preferences.value("qfieldCloudAuthcfg")
+
+        if not authcfg or authcfg == "":
+            return QgsAuthMethodConfig()
+
         cfg = QgsAuthMethodConfig()
         auth_manager.loadAuthenticationConfig(authcfg, cfg, True)
 
         return cfg
 
     def set_auth(self, url, **kwargs: str) -> None:
+        if self.auth_method != CloudAuthMethod.CREDENTIALS:
```

Comment:
I would change that to `assert`, as far as I see the caller already checks if it is the right auth method.